jdaviz/configs/default/plugins/model_fitting/tests/test_plugin.py

```diff
@@ -0,0 +1,93 @@
+"""
+Tests the features of the Model Fitting Plugin (Selecting model parameters, adding models, etc.)
+This does NOT test the actual fitting self (see test_fitting.py for that)
+"""
+
+import uuid
+
+import pytest
+
+from jdaviz.configs.default.plugins.model_fitting.initializers import MODELS
+
+
+def test_default_model_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_data(spectrum1d)
+    modelfit_plugin = specviz_helper.app.get_tray_item_from_name('g-model-fitting')
+    # By default, the spectral region should be the entire spectrum
+    assert modelfit_plugin.spectral_subset_selected == "Entire Spectrum"
+
+    for model in MODELS:
+        # Check that the auto label is set correctly (or at least the first character matches)
+        # BlackBody and Polynomial labels behave differently, so check only the first character
+        modelfit_plugin.comp_selected = model
+        assert modelfit_plugin.comp_label[0] == model[0]
+
+        # Test component label increments by default
+        previous_label = modelfit_plugin.comp_label
+        modelfit_plugin.vue_add_model()
+        assert modelfit_plugin.comp_label == previous_label + "_1"
+
+    assert len(modelfit_plugin.component_models) == len(MODELS)
+
+    # Test that default equation adds all components together
+    # Make sure to remove the trailing '+' sign!
+    assert (
+        modelfit_plugin.model_equation
+        == "+".join(param["id"] for param in modelfit_plugin.component_models)
+    )
+
+
+def test_custom_model_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_data(spectrum1d)
+    modelfit_plugin = specviz_helper.app.get_tray_item_from_name('g-model-fitting')
+
+    for model in MODELS:
+        # Add one of each model with a unique name
+        modelfit_plugin.comp_selected = model
+        modelfit_plugin.comp_label = uuid.uuid4().hex
```

Comment:
I'd prefer a label that let us see what model is what when something goes wrong. "uuid" is not very human-readable.
```suggestion
    for i, model in enumerate(MODELS):
        # Add one of each model with a unique name
        modelfit_plugin.comp_selected = model
        modelfit_plugin.comp_label = f"test_model_{i}"
```

Reply:
My general testing strategy is to break things into two camps: determinant or variable. If it's determinant, then totally agree, it should be clear what's going on. If it's variable, the argument should also be variable to try and begin to account for all the different possibilities (i.e. what the user could possibly put in).

As an example for something like this, by changing the label to something fixed, our coverage would miss any behavior that is "label dependent" right? (I'm thinking string length, hardcoded assumptions, etc.). Even taking a `uuid` to the end of it would still miss behavior that is dependent on a label "starting with X".

That being said, I'm not strongly attached to this, so if readability (and thus maintainability) is worth this trade off for you, I'm happy to satisfy it! I'm willing to be told I'm being too paranoid 😉

Reply:
I get your point but I don't think it is that applicable here. We have been using uuid when user does not give label and that has been tested at least in Imviz parser. I don't see why we need to worry about randomized label in a model fit test. IMHO it is more important that we can easily sniff out from log which model broke a test going forward. If you are passionate about randomizing input, those tests should be a separate thing and probably using a helper package like [hypothesis](https://hypothesis.readthedocs.io/en/latest/quickstart.html).